src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js'
+
 
 const App = () => {
+  const localUser = chatMessages[0].sender
+  const remoteUser = chatMessages[1].sender
+
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const onUpdateChat = (updatedChat) => {
+    const messages = chatData.map(chat => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(messages);
```

Comment:
👀 Since the next version of the chat data depends on the current version, prefer using the callback style of setChatMessagData:
```javascript
setTextMessage(prevEntries => ( 
    // logic to use current chats to make a new copy with the desired liked status flipped (the map logic above) 
    // return the new chats to update the state 
))
```

You can see an example of how we did it in Flasky [here](https://github.com/Ada-C19/flasky-front-end/blob/zoisite/src/App.js#L79-L84)